library/src/scripts/layout/LanguagesDropDown.tsx

```diff
@@ -6,67 +6,99 @@
-    selected: any;
     className?: string;
     buttonClassName?: string;
     buttonBaseClass?: ButtonTypes;
     renderLeft?: boolean;
     openAsModal?: boolean;
+    currentLocale?: string;
+    dateUpdated?: string;
 }
 
 /**
  * Implements "other languages" DropDown for articles.
  */
 export default class LanguagesDropDown extends React.Component<ILanguageDropDownProps, IState> {
     public render() {
-        const showPicker = this.props.children && this.props.children.length > 1;
-        if (showPicker) {
-            let foundIndex = false;
-            const processedChildren = this.props.children.map(language => {
-                const selected = language.lang === this.props.selected;
-                language.selected = selected;
-                if (selected) {
-                    foundIndex = selected;
-                }
-                return language;
-            });
-            if (!foundIndex) {
-                processedChildren[0].selected = true;
-            }
-            return (
-                <SelectBox
-                    describedBy={this.props.titleID!}
-                    label={!this.props.titleID ? t("Locale") : null}
-                    widthOfParent={!!this.props.widthOfParent}
-                    className={classNames("languagesDropDown", this.props.className)}
-                    renderLeft={this.props.renderLeft}
-                    buttonClassName={this.props.buttonClassName}
-                    buttonBaseClass={this.props.buttonBaseClass}
-                    openAsModal={this.props.openAsModal}
-                >
-                    {processedChildren}
-                </SelectBox>
-            );
-        } else {
+        const classes = selectBoxClasses();
+        const showPicker = this.props.data && this.props.data.length > 1;
+        if (!showPicker) {
             return null;
         }
+
+        let selectedIndex = 0;
+        const selectBoxItems: ISelectBoxItem[] = this.props.data.map((data, index) => {
+            const isSelected = data.locale === this.props.currentLocale;
+            if (isSelected) {
+                selectedIndex = index;
+            }
+            return {
+                selected: isSelected,
+                name: data.locale,
+                content: (
+                    <>
+                        <ToolTip
+                            label={`This article was editied in its source locale on ${(
+                                <DateTime timestamp={this.props.dateUpdated} />
+                            )}. Edit this article to update its translation and clear this meesage.`}
+                        >
+                            <span>
+                                <LocaleDisplayer displayLocale={data.locale} localeContent={data.locale} />
+                                {data.translationStatus === "not-translated" && (
+                                    <AlertIcon className={"selectBox-selectedIcon"} />
```

Comment:
We have a bunch of terminology here related to articles, but this component is not part of knowledge base.

I think we need to move this all 1 level higher into the KB `<OtherLanguages />` component.